sql/commitment.sql

```diff
@@ -0,0 +1,375 @@
+		plan_comm_date text,
+		project_description text,
+		commitment_description text,
+		commitment_code char(4),
+		typ_c char(3),
+		typ_c_name text,
+		plannedcommit_ccnonexempt numeric,
+		plannedcommit_ccexempt numeric,
+		plannedcommit_citycost numeric,
+		plannedcommit_nccstate numeric,
+		plannedcommit_nccfederal numeric,
+		plannedcommit_nccother numeric,
+		plannedcommit_noncitycost numeric,
+		plannedcommit_total numeric
+	);
+	
+	CREATE TABLE project_source (
+		ccp_version text,
+		m_a_proj_id text,
+		m_agency_acro text,
+		m_agency char(3),
+		m_agency_name text,
+		description text,
+		proj_id text,
+		min_date date,
+		max_date date,
+		type_category text,
+		plannedcommit_ccnonexempt numeric,
+		plannedcommit_ccexempt numeric,
+		plannedcommit_citycost numeric,
+		plannedcommit_nccstate numeric,
+		plannedcommit_nccfederal numeric,
+		plannedcommit_nccother numeric,
+		plannedcommit_noncitycost numeric,
+		plannedcommit_total numeric,
+		adopt_ccnonexempt numeric,
+		adopt_ccexempt numeric,
+		adopt_citycost numeric,
+		adopt_nccstate numeric,
+		adopt_nccfederal numeric,
+		adopt_nccother numeric,
+		adopt_noncitycost numeric,
+		adopt_total numeric,
+		allocate_ccnonexempt numeric,
+		allocate_ccexempt numeric,
+		allocate_citycost numeric,
+		allocate_nccstate numeric,
+		allocate_nccfederal numeric,
+		allocate_nccother numeric,
+		allocate_noncitycost numeric,
+		allocate_total numeric,
+		commit_ccnonexempt numeric,
+		commit_ccexempt numeric,
+		commit_citycost numeric,
+		commit_nccstate numeric,
+		commit_nccfederal numeric,
+		commit_nccother numeric,
+		commit_noncitycost numeric,
+		commit_total numeric,
+		spent_ccnonexempt numeric,
+		spent_ccexempt numeric,
+		spent_citycost numeric,
+		spent_nccstate numeric,
+		spent_nccfederal numeric,
+		spent_nccother numeric,
+		spent_noncitycost numeric,
+		spent_total numeric,
+		spent_total_checkbooknyc numeric
+	);
+	
+COMMIT;
+
+-- Populate source tables
+BEGIN;
+	COPY project_source FROM 'cpdb_projects_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+BEGIN;
+	COPY commitment_source FROM 'cpdb_planned_commitments_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+-- Structure of target tables
+-- Right now, we are using data as primary keys. We should evaluate when they should be replaced with separate ids
+BEGIN;
+	CREATE TABLE managing_code (
+		id char(3) PRIMARY KEY
+	);
+
+	-- Should likely create an id distinct from initials
+	CREATE TABLE agency (
+		initials text PRIMARY KEY,
+		name text
+	);
+	
+ -- likely unnecessary as a table because the project table already captures both its managing code and its managing agency 
+	-- CREATE TABLE agency_managing_code (
+	-- 	agency_initials text references agency(initials),
+	-- 	managing_code_id char(3) references managing_code(id)
+	-- );
+	
+	CREATE TYPE project_category AS ENUM (
+		'Fixed Asset',
+		'Lump Sum',
+		'ITT, Vehicles, and Equipment'
+	);
+		
+	CREATE TABLE project (
+		PRIMARY KEY (managing_code, id),
```

Comment:
It's simpler from one perspective but not simpler overall. Dropping the managing code table also means dropping the managing codes column from the project and commitment tables. This obfuscates which projects belong to the same managing code. When we query that relationship, we need to manually check the first three digits of the maprojid. Postgres supports this query. However, the idiomatic approach is to split the data into its own table.

Reply:
If you're looking for projects with the same managing code, isn't that basically saying "projects that belong to the same agency"? In my approach, I was thinking we could still accomplish that with the foreign key to the agencies table.

Reply:
If the reality of managing codes matched their intention, then managing codes and agencies would indeed have a 1 to 1 mapping. Under this model, we would put managing codes as a column within the agency table and the data model would be in harmony with reality.

In practice, there are two exceptions that buck this data model. First we don't always know the agency associated with the managing code. It would be misleading to say that two projects associated with "unknown" agencies are associated with "the same" agency. Second, only showing that a project is associated with "Brooklyn Naval Yard", "Gov. Island Trust", or "EDC" would miss that these three 'agencies' are associated with the same managing code and each other.

Reply:
Following up after talking about this IRL - I'm sold on how you're handling managing agency codes. Having "agency code" as a first class citizen in the DB nicely models how they relate to other entities.

As for "unknown", I'm leaning slightly towards having orphaned projects rather than assigning them to an "unknown agency" but there are practical and semantic pros and cons to each approach so I'll defer to you.

Reply:
I'm defacto deferring to how folks further up the data-processing chain handle "unknown" conditions - denoting it as a special "UK agency"